### PR TITLE
Fixes #46: Filters for `DataPath`s

### LINKED ISSUE #46: `DataPath`: Filters that stop a discovered value from being considered an instance of the respective property
Sometimes, a tracker will transmit something like `idfa: none`. Currently, this would be detected as an IDFA transmission, which is obviously wrong.

I think we'll need filters to prevent that. Maybe it's easiest if we can either specify a positive (`onlyIf: \d+`) or negative (`notIf: none`) regex?

### PR DESCRIPTION
**Pull request by baltpeter**
This can be squashed.

### CODE REVIEW

**Comment by baltpeter** [comment]
This is now based on https://github.com/tweaselORG/TrackHAR/pull/87, which needs to be merged first.

File: src/index.ts [hunk 1]
```diff
@@ -360,6 +364,17 @@ export const processRequest = (
...
         .flat();
+    /**
+     * Transform a value transmitted by a tracker, which can be any type, into a string, choosing the most
+     * "natural-feeling" representation to make it filterable.
+     *
+     * For example, objects and arrays will be returned as JSON strings, whereas strings will be returned as-is (and
+     * thus notably without enclosing quotes as `JSON.stringify` would add).
+     */
+    const stringify = (v: unknown) =>
+        typeof v === 'bigint' || typeof v === 'string' || typeof v === 'symbol' || typeof v === 'function'
```
Comment (zner0L) [inline src/index.ts:375]: Where is this list coming from? Can `JSON.stringify()` handle all other JS types?

**Review by zner0L — COMMENTED** [review]
I like the filters for each `DataPath`, but I also think that there are some values which always signify an empty result, such as `none`, `unknown`or `null`. I think we should filter for these as a default to be safe, maybe with the option overwrite this using the `onlyIf` option.

File: src/index.ts (hunk 1 above)
Comment (baltpeter) [reply src/index.ts:375]: Manual testing by me. Yep.

**Comment by baltpeter** [comment]
> I like the filters for each `DataPath`, but I also think that there are some values which always signify an empty result, such as `none`, `unknown`or `null`. I think we should filter for these as a default to be safe, maybe with the option overwrite this using the `onlyIf` option.

Yeah, that's a good idea. I'll implement it. I don't think implementing overrides would be practical, but I really don't think they should be necessary, either.

File: src/index.ts [hunk 2]
```diff
@@ -394,6 +394,8 @@ export const processRequest = (
                             'none',
                             'n/a',
                             '00000000-0000-0000-0000-000000000000',
+                            '""',
+                            "''",
```
Comment (zner0L) [inline src/index.ts:398]: Please update the README accordingly.

**Review by zner0L — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (baltpeter) [reply src/index.ts:398]: But the README already mentions the behaviour? https://github.com/tweaselORG/TrackHAR/pull/83/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R255

Or, if you're talking specifically about the quotes, it deliberately wasn't a complete list before this commit, either.

**Review by zner0L — DISMISSED** [review]

**Pull request merged**